### ISSUE TITLE
Pin Behaviour Bug

### ISSUE THREAD

**Issue by BonzerKitten7** [Scirra/Construct-3-bugs#3271]
## Problem description

I noticed that a bug that was not present in my game before, is now affecting players, because of the recent C3 updates. For some reason the Pin Behaviour seems to have a "lower priority" in the event sheet, because even if I pin a sprite to another sprite and then try to move the main object in the same tick, there will be a "offset" between them. This is a bug because this is not how things were working before, I had to implement a bit of "wait" to fix, which is very annoying.

## Attach a .c3p

[Pin Bug.zip](https://github.com/Scirra/Construct-3-bugs/files/3542491/Pin.Bug.zip)

## Steps to reproduce

1. Just start the project.

## Observed result

Offset between both objects.

## Expected result

No offset between both objects.

## More details

**Affected browsers/platforms:** 

**First affected release:** 

## System details

<details><summary>View details</summary>

PASTE HERE

</details>

**Comment by QuasarX** [comment]
[PinBugFix.zip](https://github.com/Scirra/Construct-3-bugs/files/3544914/PinBugFix.zip)

**Comment by AshleyScirra** [comment]
The problem is there is a dependency here on the particular order the engine fires events, which is currently not specified and can vary depending on different aspects of your project.

As it is, it fires *On created* for spr_collision first, which moves it to the right. Then it fires *On created* for spr_graphics, which pins it to spr_collision which is no longer overlapping it.

It appears you expected it to fire *On created* the opposite way round, but there's no particular reason for this to be the "correct" order, and indeed even if it was changed to that order, you simply need to swap the actions around and it's "broken" again.

Basically you can't rely on any particular ordering of *On created* triggers. You just need to adjust the events to initialise them in a specific order (e.g. using actions in an *On start of layout* event).

**Comment by AshleyScirra** [comment]
Events do run in top-to-bottom order, but the ordering here is the ordering of object instances triggering *On created*. You could use any ordering scheme at all for that, but as I explained, it won't fundamentally solve this problem, since the same problem can happen if you then just adjust the order of the events instead. So there's no point trying to change or define a specific ordering scheme.
